packages/repository/src/__tests__/integration/repositories/resolve-has-many-metadata.integration.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  Entity,
+  HasManyDefinition,
+  ModelDefinition,
+  RelationType,
+} from '../../..';
+import {resolveHasManyMetadata} from '../../../relations/has-many/has-many.helpers';
+
+describe('keyTo and keyFrom with resolveHasManyMetadata', () => {
+  it('resolves metadata using keyTo and keyFrom', () => {
+    const meta = resolveHasManyMetadata(Category.definition.relations[
+      'products'
+    ] as HasManyDefinition);
+
+    expect(meta).to.eql({
+      name: 'products',
+      type: 'hasMany',
+      targetsMany: true,
+      source: Category,
+      keyFrom: 'id',
+      target: () => Product,
+      keyTo: 'categoryId',
+    });
+  });
+
+  it('resolves metadata using keyTo, but not keyFrom', () => {
+    const meta = resolveHasManyMetadata(Category.definition.relations[
+      'items'
+    ] as HasManyDefinition);
+
+    expect(meta).to.not.have.property('keyFrom');
+
+    expect(meta).to.eql({
+      name: 'items',
+      type: 'hasMany',
+      targetsMany: true,
+      source: Category,
+      target: () => Item,
+      keyTo: 'categoryId',
+    });
+  });
+
+  it('infers keyTo if is it not provided', () => {
+    const meta = resolveHasManyMetadata(Category.definition.relations[
+      'things'
+    ] as HasManyDefinition);
+
+    expect(meta).to.eql({
+      name: 'things',
+      type: 'hasMany',
+      targetsMany: true,
+      source: Category,
+      keyFrom: 'id',
+      target: () => Thing,
+      keyTo: 'categoryId',
+    });
+  });
+
+  it('throws if both keyFrom and keyTo are not provided', async () => {
```

Comment:
Does it thorw error because both`keyFrom` and `keyTo` are not provided and the default name for `keyTo` is not defined in `Category` as well? 
I know we probably cover the default name somewhere else. Can we still add it here? I was a bit confused at the beginning. 
 Besides this, the rest look good to me 👍

Reply:
Ah, when I was testing it before I only added `categoryId` to the class but not to the model definition, but I see that when I add it now, it works. Let me modify the tests 👍